semgrep/semgrep/evaluation.py

```diff
@@ -263,34 +265,25 @@ def _evaluate_single_expression(
     else:
         raise UnknownOperatorError(f"unknown operator {expression.operator}")
 
-    metavars_for_patterns = get_metavar_debugging_info(
```

Comment:
These two calls were only called in two places: `_evaluate_single_expression` and `_evaluate_expression`. We can turn this into a single location as was done in this change by pulling the functionality up a level in `_evaluate_expression`. This makes for a much cleaner solution.